src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/SasQueryParamsGenerator.java

```diff
@@ -24,30 +24,31 @@
 
 @EnableConfigurationProperties(AccessTokenProperties.class)
 @Service
-public class SasTokenGeneratorService {
+public class SasQueryParamsGenerator {
 
-    private static final Logger log = LoggerFactory.getLogger(SasTokenGeneratorService.class);
+    private static final Logger log = LoggerFactory.getLogger(SasQueryParamsGenerator.class);
 
     private final CloudBlobClient cloudBlobClient;
     private final AccessTokenProperties accessTokenProperties;
 
-    public SasTokenGeneratorService(
+    public SasQueryParamsGenerator(
         CloudBlobClient cloudBlobClient,
         AccessTokenProperties accessTokenProperties
     ) {
         this.cloudBlobClient = cloudBlobClient;
         this.accessTokenProperties = accessTokenProperties;
     }
 
-    public String generateSasToken(String serviceName) {
+    public String generateSasQueryParams(String serviceName) {
 
         log.info("SAS Token request received for service {} ", serviceName);
 
         try {
-            return cloudBlobClient
+            String token = cloudBlobClient
                 .getContainerReference(serviceName)
                 .generateSharedAccessSignature(createSharedAccessPolicy(serviceName), null);
 
+            return token.charAt(0) == '?' ? token : "?" + token;
```

Comment:
Are we covering test cases for both the scenarios ?

Reply:
all tests written around sas token generator/controller are not using any mocks and token just comes back without `?`. Don't see the benefit of adding tests introducing mocks just to cover the branch despite the fact I'm in favour of having everything covered